core/src/main/java/org/apache/gravitino/authorization/AuthorizationRequestContext.java

```diff
@@ -77,11 +77,9 @@ public boolean authorizeDeny(
   }
 
   public void loadRole(Runnable runnable) {
-    if (hasLoadRole.get()) {
-      return;
+    if (hasLoadRole.compareAndSet(false, true)) {
```

Comment:
This could cause some threads to skip loadRole under concurrent conditions, leading to incorrect results.